### ISSUE TITLE
Support for SAML tokens

### ISSUE THREAD

**Issue by pelikhan** [gr2m/octokit-plugin-create-pull-request#67]
Organization may requires SAML token (like Microsoft) and it seems that the fork bailout does not detect this properly.
Would it be possible force a fork -- or -- support SAML token?

Thanks for this tool, it's great.

**Comment by gr2m** [comment]
I think supporting SAML tokens would be great! I don't have a way to test it myself, I'd need your help. What's the error message you are currently getting? Could you log the `error` object and share the output here? Just make sure to remove sensitive information from the output first

**Comment by pelikhan** [comment]
Console shows this:
![image](https://user-images.githubusercontent.com/4175913/105543681-5f934b00-5cfa-11eb-8911-972c793b0cdd.png)

**Comment by pelikhan** [comment]
Response is 403
```
{
  "message": "Resource protected by organization SAML enforcement. You must grant your Personal Access token access to this organization.",
  "documentation_url": "https://docs.github.com/articles/authenticating-to-a-github-organization-with-saml-single-sign-on/"
}
```

**Comment by gr2m** [comment]
My guess is that the `GET /repos/{owner}/{repo}` request fails

https://github.com/gr2m/octokit-plugin-create-pull-request/blob/f2abf7fe1acdc07685306ee53069778e12080629/src/compose-create-pull-request.ts#L32-L39

We should catch the error and check if its message includes `/\bSAML\b`. If it does, we have to run the same code that we currently run when the user does not have write access to the repository

https://github.com/gr2m/octokit-plugin-create-pull-request/blob/f2abf7fe1acdc07685306ee53069778e12080629/src/compose-create-pull-request.ts#L55-L78

Would you like to send a pull request with the required change? It'd be easier for you as you can test it against a real SAML protected repository. I can help you through the process if you have any questions

**Comment by pelikhan** [comment]
how do you test locally? npm link?

**Comment by gr2m** [comment]
Yes. After you checkout this repository, you need to run `npm install` and `npm build`. After each change, you need to run `npm build` again. Then link the `pkg/` subfolder, which will have the contents that is usually published to npm, it will look like this: https://unpkg.com/browse/octokit-plugin-create-pull-request@3.9.2

**Comment by pelikhan** [comment]
I'll get to it eventually...
